builtin/credential/userpass/path_users.go

```diff
@@ -167,19 +167,18 @@ func (b *backend) userCreateUpdate(ctx context.Context, req *logical.Request, d
 		userEntry.Policies = policyutil.ParsePolicies(policiesRaw)
 	}
 
-	ttlStr := userEntry.TTL.String()
-	if ttlStrRaw, ok := d.GetOk("ttl"); ok {
-		ttlStr = ttlStrRaw.(string)
+	ttl, ok := d.GetOk("ttl")
+	if ok {
+		userEntry.TTL = time.Duration(ttl.(int)) * time.Second
+	} else if req.Operation == logical.CreateOperation {
```

Comment:
We don't need to do this -- there is no default set so it will be 0 anyways, and 0 will just mean to have core use the sysview values.

Reply:
In fact I think you don't need GetOk either, just read directly and it'll be zero or not.

Reply:
We will need to distinguish between the CreateOperation and UpdateOperation. So, GetOk is required. Removed the `else if` block though.